src/mem_table.rs

```diff
@@ -113,12 +113,17 @@ pub(crate) struct WritableKVTable {
 }
 
 pub(crate) struct ImmutableMemtable {
-    /// The last WAL ID of this IMM. This is used to determine the starting position of
-    /// WAL replay during recovery. After an IMM is flushed to L0, we do not need to care
-    /// about the earlier WALs which produced this IMM, all we need to know is the last
+    /// The recent flushed WAL ID when this IMM is freezed. This is used to determine the starting
+    /// position of WAL replay during recovery. After an IMM is flushed to L0, we do not need to
+    /// care about the earlier WALs which produced this IMM, all we need to know is the recent
     /// WAL ID of the last L0 compacted.
-    last_wal_id: u64,
+    ///
+    /// Please note that this recent flushed WAL ID might not exactly match the last WAL ID that
```

Comment:
Just to clarify my undestanding, when running the WAL recovery right now, I believe we start at `last_l0_recent_flushed_wal_sst_id + 1`. This behavior will remain, right? The `last_seq` filtering will be continue to be applied starting in the WAL _after_ `last_l0_recent_flushed_wal_sst_id + 1`, right?

Reply:
yes, the `last_seq` filter will happens from the wal at `last_l0_recent_flushed_wal_sst_id + 1`, and this allows us less to be strict on the correctness of `last_l0_recent_flushed_wal_sst_id` (it's ok to have a stale value, as we will skip the entries which had already applied in l0 during the replay)